ln-resource-mgr/src/forward_manager.rs

```diff
@@ -276,18 +285,26 @@ impl ReputationManager for ForwardManager {
                 let congestion_liquidity_amount =
                     capacity_msat * self.params.congestion_liquidity_portion as u64 / 100;
 
+                let general_bucket = BucketParameters {
+                    slot_count: general_slot_count,
+                    liquidity_msat: general_liquidity_amount,
+                };
+
+                let congestion_bucket = BucketParameters {
+                    slot_count: congestion_slot_count,
+                    liquidity_msat: congestion_liquidity_amount,
+                };
+
                 v.insert(TrackedChannel {
-                    incoming_direction: IncomingChannel::new(self.params.reputation_params),
+                    incoming_direction: IncomingChannel::new(
+                        self.params.reputation_params,
+                        general_bucket.clone(),
+                        congestion_bucket.clone(),
+                    ),
```

Comment:
We don't need a bucket on the incoming channel, this is the one asymmetry that we have  in all this - we can look at reputation in the outgoing or incoming direction but we're only allocating resources on the outgoing side (since we've already accepted the incoming HTLC).

I don't think that this is clearly written down anywhere, and I probably didn't do a good job at explaining it - my bad!